frontend/src/Game/Match.tsx

```diff
@@ -80,17 +159,46 @@ function Game(): ReactElement {
       <div id="rightScore">{p2Score}</div>
       <div id="gameDiv"></div>
       <Ball pBall={pBall} />
+      <Paddle paddle={leftPaddle} />
+      <Paddle paddle={rightPaddle} />
     </div>
   )
 }
 
 export function Match(): ReactElement {
+  const [score, setScore] = useState<number>(0)
+  const handleScoreChange = (): void => {
+    setScore((score) => {
+      console.log(score + 1)
+      return score + 1
+    })
+  }
+
+  useEffect(() => {
+    if (score === 10) {
+      console.log('GAME SET')
+    }
+  }, [score])
+
+  const handleOnKeyDonw = (e: KeyboardEvent): void => {
+    keydown = e.code
+  }
+
+  const handleOnKeyUp = (): void => {
+    keydown = ''
+  }
+
+  window.addEventListener('keydown', handleOnKeyDonw)
+  window.addEventListener('keyup', handleOnKeyUp)
```

Comment:
毎回addEventListenerしなくても、useEffectをつかって初期レンダー時にだけ登録すればいいかも？
手元では動きましたがちょっと自信ないです

```suggestion
  useEffect(() => {
    const handleOnKeyDown = (e: KeyboardEvent): void => {
      keydown = e.code
    }

    const handleOnKeyUp = (): void => {
      keydown = ''
    }

    window.addEventListener('keydown', handleOnKeyDown)
    window.addEventListener('keyup', handleOnKeyUp)
  }, [])
```

Reply:
確かに！初期レンダー時だけで十分そうですね。変更します